### DISCUSSION TITLE
Has anyone tried running a script against the agent via the API?

### DISCUSSION THREAD

**Discussion by AshleyJackson** [amidaware/tacticalrmm#1431] [General]
The swagger docs are limited in explaining if any parameters are required for the below api call
![image](https://user-images.githubusercontent.com/12793989/217161504-c34d36cb-ee6f-4d64-85f8-6223e8884181.png)

In Postman it looks like that specific call (with and without any other paramters) just spits back a 500 Server Error
![image](https://user-images.githubusercontent.com/12793989/217161713-e20e55c1-765f-49d2-89a6-afedfd36b797.png)
![image](https://user-images.githubusercontent.com/12793989/217161768-7c1c0c2a-a886-4416-8631-5d87d2cc2f06.png)

And on top of this, I can't find any live logs of this error happening
(NOTE: I am running in docker, but I don't think that should effect the api calls as other api calls work fine.)

**Comment by wh1te909** [comment]
Please read https://docs.tacticalrmm.com/functions/api/ you need to open your brower's dev tools, perform the action from the web UI and you will see exactly the payload you need

**Reply by saulens22** [reply]
The API documentation currently is very crude... Would it be possible to list all API endpoints in documentation? It should be possible using `django-extensions` command `show_urls`.

https://django-extensions.readthedocs.io/en/latest/command_extensions.html?highlight=show_urls#command-extensions

**Reply by wh1te909** [reply]
enable swagger and you will see all the endpoints

**Comment by AshleyJackson** [comment]
Hi @wh1te909.
Thanks for the response.

I've tried that and the error that comes back seems to be the exact same error that I was receiving in Postman.

![image](https://user-images.githubusercontent.com/12793989/217410694-2e5097f8-09d3-44af-a9be-cc286e59de71.png)


After runnig the above api I looked through the logs for the environment and there were zero errors.
And just to make sure it wasn't the Agent ID being wrong, I tested the agent and successfully performed a reboot of the device via api.

Are there any debug conditions I can enable for the environment to get more details when attempting this api query?

Thanks.

**Reply by wh1te909** [reply]
500 errors are logged in `django_debug.log` search docs for this. Also I don't see you including any payload. You need to send a payload with the post request, like the id of the script, the timeout, script args, env vars etc. When I said to open browser dev tools and do the request, I meant from the tacticalrmm frontend, not from postman. Run an actual real script on the agent and then in dev tools you will see the exact json payload being sent.

**Reply by AshleyJackson** [reply]
Ah, gotcha.
Just wasn't seeing any details about what payload to send in the swagger docs.

I'll try testing the script payload now and get back to you.

Thanks in advance.

**Reply by AshleyJackson** [reply]
So that ended up working and I was able to run scripts via API.
Thank you for that.

The follow up issue I am witnessing is that the Gateway is timing out after 60 seconds, when the script timeout is set to 120 seconds.

![image](https://user-images.githubusercontent.com/12793989/218359003-3888854e-4185-4216-ab74-f5dcfe1c00de.png)

Is there some configuration that I am missing to prevent the gateway from timing out like this?

Thanks for your time.

**Reply by wh1te909** [reply]
check nginx config

**Reply by wh1te909** [reply]
the nginx timeout set during install is 300 seconds unless you changed something. if you need longer than 300 seconds you should use the fire and forget method, there is no limit on the timeout for that.
